### PR TITLE
feat: add ability to pass `includesArtworkID` arg to `collectionsConnection`

### PR DESCRIPTION
**Pull request by dimatretyak**
Jira ticket: [FX-4784](https://artsyproduct.atlassian.net/browse/FX-4784)

### Demo
https://github.com/artsy/metaphysics/assets/3513494/9ac28f2d-4dca-4219-bb87-248b0d24d71d

[FX-4784]: https://artsyproduct.atlassian.net/browse/FX-4784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/schema/v2/me/__tests__/collectionsConnection.test.ts [hunk 1]
```diff
@@ -32,17 +32,51 @@ describe("collectionsConnection", () => {
...
+      await runAuthenticatedQuery(query, context)
 
-    expect(context.collectionsLoader as jest.Mock).toHaveBeenCalledWith({
-      user_id: "user-42",
-      private: true,
-      saves: true,
-      sort: "-created_at",
-      offset: 0,
-      size: 1,
-      total_count: true,
```
Comment (anandaroop) [inline src/schema/v2/me/__tests__/collectionsConnection.test.ts:45]: **suggestion:** Honestly, I don't think adding a whole new test case here is providing much value here.

We just want to know that this resolved sends the expected payload upstream.

I would leave the `it("passes correct args to Gravity")` test as it was, and only make two small changes: 
- add another `includesArtworkID` on L11 above
- add `artwork_id: "123-abc"` to the expected payload here.

**Review by anandaroop — CHANGES_REQUESTED** [review]
Thanks! 

The change looks good, but I would recommend simplifying the test as suggested below.

File: src/schema/v2/me/__tests__/collectionsConnection.test.ts (hunk 1 above)
Comment (dimatretyak) [reply src/schema/v2/me/__tests__/collectionsConnection.test.ts:45]: Done: https://github.com/artsy/metaphysics/pull/5022/commits/663ced4abcccf7358259c2d2991c9595afbb765d

File: src/schema/v2/me/__tests__/collectionsConnection.test.ts [hunk 2]
```diff
@@ -43,6 +48,39 @@ describe("collectionsConnection", () => {
       offset: 0,
       size: 1,
       total_count: true,
+      artwork_id: "123-abc",
+    })
+  })
+
+  it("passes correct args to Gravity when `includesArtworkID` arg is specified", async () => {
+    const query = gql`
```
Comment (anandaroop) [inline src/schema/v2/me/__tests__/collectionsConnection.test.ts:56]: We just need to remove this test case now, since it's already covered above 😄

**Review by anandaroop — APPROVED** [review]

File: src/schema/v2/me/__tests__/collectionsConnection.test.ts (hunk 2 above)
Comment (dimatretyak) [reply src/schema/v2/me/__tests__/collectionsConnection.test.ts]: Addressed here: https://github.com/artsy/metaphysics/pull/5022/commits/4bb9e4d4861a1434b82f15da72fd40d6112803cb

**Pull request merged**